File: Algorithms/Searching/Binary Search/BinarySearch.py
def BinarySearch(arr,val):
    start = 0
    end=len(arr)
    mid=end/2
    while (arr[mid] != val and start < end):
        if arr[mid] > val:
            end=mid-1
        else:
            start = mid + 1
        mid = (start+end)/2
    return str(mid) if (arr[mid]==val) else '-1'


arr = [1,5,4,2,7,8,2,12,63]
value1 = 7
value3 = 1
value2 = 45
print(str(value1) + ' is found at '+ BinarySearch(arr,value1))
print(str(value2) + ' is found at '+ BinarySearch(arr,value2))
print(str(value3) + ' is found at '+ BinarySearch(arr,value3))

File: Algorithms/Searching/Binary Search/BinarySearch.js
function BinarySearch(arr,val){
    var start = 0;
    var end = arr.length;
    var mid = Math.floor(end/2);

    while(arr[mid]!= val && start < end){
        if(arr[mid] < val){
            start = mid + 1;
        }else{
            end = mid - 1;
        }
        mid = Math.floor((start+end)/2)
    }
    return (arr[mid]==val)? mid: -1;
}

var arr = [1,2,4,5,7,8,12,63];
var value1 = 5,
value2 = 54,
value3 = 4;
console.log(`${value1} is at ${BinarySearch(arr,value1)}`);
console.log(`${value2} is at ${BinarySearch(arr,value2)}`);
console.log(`${value3} is at ${BinarySearch(arr,value3)}`);